src/components/WorkPoolCreateForm.vue

```diff
@@ -31,26 +39,47 @@
 
 <script lang="ts" setup>
   import { showToast } from '@prefecthq/prefect-design'
-  import { useValidation, useValidationObserver, ValidationRule } from '@prefecthq/vue-compositions'
-  import { ref } from 'vue'
+  import { useSubscription, useValidation, useValidationObserver, ValidationRule } from '@prefecthq/vue-compositions'
+  import { computed, ref } from 'vue'
   import { useRouter } from 'vue-router'
-  import { SubmitButton } from '@/components'
-  import { useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
+  import { SubmitButton, WorkPoolTypeSelect, SchemaFormFieldsWithValues } from '@/components'
+  import { useCan, useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
   import { localization } from '@/localization'
+  import { WorkPoolCreate } from '@/models'
+  import { mapper } from '@/services'
+  import { Schema } from '@/types'
 
   const api = useWorkspaceApi()
+  const can = useCan()
   const router = useRouter()
   const routes = useWorkspaceRoutes()
 
   const { validate, pending } = useValidationObserver()
 
   const name = ref<string>()
   const description = ref<string>()
-  // Once types feature is implemented by backend, this should be changed to
-  // const type = ref<string>()
-  const type = ref<string>('prefect-agent')
+  const type = ref<string>()
   const concurrencyLimit = ref<number>()
 
+  const workersCollectionSubscription = useSubscription(api.collections.getWorkerCollection, [])
+  const workersCollectionItems = computed(() => workersCollectionSubscription.response)
+  const baseJobConfigs = computed(() => {
+    return workersCollectionItems.value?.find((item) => item.type === type.value)?.defaultBaseJobConfiguration ?? {}
+  })
+
+  const schema = computed<Schema>(() => mapper.map('SchemaResponse', baseJobConfigs.value.variables ?? {}, 'Schema'))
+  const parameters = ref()
+
+  const testSchema = computed<Schema>(() => {
+    return { ...schema.value, type: 'object' }
+  })
```

Comment:
@desertaxle just calling out that this `testSchema` was for debugging. I couldn't figure out why the schema wasn't rendering and turned out to be because the schema fields component was expecting the `object` type. Not sure if that's a bug in the schemes themselves being used here or if the component should know that anything with `properties` is an `object`.